actors/miner/src/lib.rs

```diff
@@ -2995,6 +2992,197 @@ impl Actor {
         Ok(())
     }
 
+    fn move_partitions(rt: &impl Runtime, params: MovePartitionsParams) -> Result<(), ActorError> {
+        if params.from_deadline == params.to_deadline {
+            return Err(actor_error!(illegal_argument, "from_deadline == to_deadline"));
+        }
+        let policy = rt.policy();
+        if params.from_deadline >= policy.wpost_period_deadlines
+            || params.to_deadline >= policy.wpost_period_deadlines
+        {
+            return Err(actor_error!(
+                illegal_argument,
+                "invalid param, from_deadline: {}, to_deadline: {}",
+                params.from_deadline,
+                params.to_deadline
+            ));
+        }
+        if params.partitions.is_empty() {
+            return Err(actor_error!(illegal_argument, "empty partitions not allowed"));
+        }
+
+        rt.transaction(|state: &mut State, rt| {
+            let info = get_miner_info(rt.store(), state)?;
+
+            rt.validate_immediate_caller_is(
+                info.control_addresses.iter().chain(&[info.worker, info.owner]),
+            )?;
+
+            let store = rt.store();
+            let current_deadline = state.deadline_info(policy, rt.curr_epoch());
+            let mut deadlines = state
+                .load_deadlines(store)
+                .context_code(ExitCode::USR_ILLEGAL_STATE, "failed to load deadlines")?;
+
+            deadline_available_for_move(
+                policy,
+                params.from_deadline,
+                params.to_deadline,
+                &current_deadline,
+            )
+            .context_code(
+                ExitCode::USR_FORBIDDEN,
+                "conditions not satisfied for deadline_available_for_move",
+            )?;
+
+            let mut from_deadline =
+                deadlines.load_deadline(policy, store, params.from_deadline).context_code(
+                    ExitCode::USR_ILLEGAL_STATE,
+                    format!("failed to load deadline {}", params.from_deadline),
+                )?;
+            // only try to do synchronous Window Post verification if the from deadline is in dispute window
```

Comment:
The synchronous (PoST) here just means that `verify_windowed_post` will be called right in the current transaction instead of asynchronously(delayed).